src/configuration/utils.ts

```diff
@@ -0,0 +1,15 @@
+import { PermissionEnum } from "@saleor/types/globalTypes";
+import { IntlShape } from "react-intl";
+
+import { createConfigurationMenu } from ".";
+
+export const getConfigMenuItemsPermissions = (
+  intl: IntlShape
+): PermissionEnum[] =>
+  createConfigurationMenu(intl)
+    .reduce(
+      (prev, next) =>
+        prev.concat(next.menuItems.map(menuItem => menuItem.permissions)),
+      []
+    )
+    .flat();
```

Comment:
This is no longer needed, you're already flattening the array in the step above.